src/header/name.rs

```diff
@@ -1760,33 +1235,31 @@ impl HeaderName {
     /// let a = HeaderName::from_static("foobar");
     /// let b = HeaderName::from_static("FOOBAR"); // This line panics!
     /// ```
-    #[allow(deprecated)]
-    pub fn from_static(src: &'static str) -> HeaderName {
-        let bytes = src.as_bytes();
-        #[allow(deprecated)]
-        let mut buf = unsafe { mem::uninitialized() };
-        match parse_hdr(bytes, &mut buf, &HEADER_CHARS_H2) {
-            Ok(hdr_name) => match hdr_name.inner {
-                Repr::Standard(std) => std.into(),
-                Repr::Custom(MaybeLower { buf: _, lower: true }) => {
-                    let val = ByteStr::from_static(src);
-                    Custom(val).into()
-                },
-                Repr::Custom(MaybeLower { buf: _, lower: false }) => {
-                    // With lower false, the string is left unchecked by
-                    // parse_hdr and must be validated manually.
-                    for &b in bytes.iter() {
-                        if HEADER_CHARS_H2[b as usize] == 0 {
-                            panic!("invalid header name")
-                        }
-                    }
+    #[allow(unconditional_panic)] // required for the panic circumvention
+    pub const fn from_static(src: &'static str) -> HeaderName {
+        let name_bytes = src.as_bytes();
+        if let Some(standard) = StandardHeader::from_bytes(name_bytes) {
+            return HeaderName{
```

Comment:
I'd think rustfmt would catch the missing space... Not that I care, but makes me wonder 🤔